man/systemd.exec.xml

```diff
@@ -1699,6 +1699,25 @@ BindReadOnlyPaths=/var/lib/systemd</programlisting>
         logging.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>RestrictFileSystems=</varname></term>
+
+        <listitem><para>Restricts the set of filesystems processes of this unit can open files on. Takes a space-separated
+        list of filesystem names. Any filesystem listed is made accessible to the unit's processes, access to filesystem
+        types not listed is prohibited (allow-listing). If the empty string is assigned, access to filesystems is not
```

Comment:
would it be possible to add the negation syntax ? 
The first use case I can think of is denying access to fat/vfat to some processes and I might not know the kind of filesystem the rootfs itself is on..

Alternatlively some some sort of AllowFilesystemsForDirectory= where we give a path and it allows whatever  filesystem that path is on could be usefull

Reply:
> would it be possible to add the negation syntax ?

The current BPF implementation doesn't support this but it shouldn't be too difficult to support. I agree this would be nice.

> Alternatlively some some sort of AllowFilesystemsForDirectory= where we give a path and it allows whatever filesystem that path is on could be usefull

Not sure this is too useful since it will only support one filesystem, the negation syntax is probably more flexible and should cover the same use case although the user would have to specify the filesystem type. We could also make it a list of directories but it feels weird to me.